src/ER/Ringdowns.js

```diff
@@ -1,20 +1,54 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
+import ApiService from '../ApiService';
 
 import Heading from '../Components/Heading';
 import RingdownCard from '../Components/RingdownCard';
 import Ringdown from '../Models/Ringdown';
 
 function Ringdowns({ ringdowns }) {
+  const [enroute, setEnroute] = useState([]);
+
   const waiting = ringdowns.filter(
     (r) => r.currentDeliveryStatus === Ringdown.Status.ARRIVED || r.currentDeliveryStatus === Ringdown.Status.OFFLOADED
   );
-  const enroute = ringdowns.filter(
-    (r) =>
-      r.currentDeliveryStatus !== Ringdown.Status.ARRIVED &&
-      r.currentDeliveryStatus !== Ringdown.Status.OFFLOADED &&
-      r.currentDeliveryStatus !== Ringdown.Status.RETURNED_TO_SERVICE
-  );
+
+  useEffect(() => {
+    // set enroute ringdowns once data comes back from server.
+    const handler = setTimeout(() => {
+      setEnroute(
+        ringdowns.filter(
+          (r) =>
+            r.currentDeliveryStatus !== Ringdown.Status.ARRIVED &&
+            r.currentDeliveryStatus !== Ringdown.Status.OFFLOADED &&
+            r.currentDeliveryStatus !== Ringdown.Status.CANCEL_ACKNOWLEDGED &&
+            r.currentDeliveryStatus !== Ringdown.Status.REDIRECT_ACKNOWLEDGED &&
+            r.currentDeliveryStatus !== Ringdown.Status.RETURNED_TO_SERVICE
+        )
+      );
+    }, 100);
+    return () => {
+      clearTimeout(handler);
+    };
+  });
+
+  function onStatusChange(rd, status) {
+    // submit to server
+    const now = new Date();
+    ApiService.ringdowns.setDeliveryStatus(rd.id, status, now);
+    // update local object for immediate feedback
+    rd.currentDeliveryStatus = status;
+    setEnroute(
+      ringdowns.filter(
+        (r) =>
```

Comment:
How about we refactor this filter function out into a reusable function outside of the component in this same file so we don't repeat this large conditional ?